java/java.file.launcher/src/org/netbeans/modules/java/file/launcher/SingleSourceFileUtil.java

```diff
@@ -68,15 +73,15 @@ static FileObject getJavaFileWithoutProjectFromLookup(Lookup lookup) {
         return null;
     }
 
-    static boolean isSingleSourceFile(FileObject fObj) {
+    public static boolean isSingleSourceFile(FileObject fObj) {
         Project p = FileOwnerQuery.getOwner(fObj);
         if (p != null || !fObj.getExt().equalsIgnoreCase("java")) { //NOI18N
             return false;
         }
         return true;
     }
 
-    static Process compileJavaSource(FileObject fileObject) {
+    public static Process compileJavaSource(FileObject fileObject) {
         FileObject javac = JavaPlatformManager.getDefault().getDefaultPlatform().findTool("javac"); //NOI18N
```

Comment:
Thinking about adding a JDK selector to the file properties of single source programs for NB 21 to decouple it from the runtime JDK. Is something like this planned? Will this still work in VSCode?

Reply:
I think I (and possibly others) are considering separating the runtime JDK and the JavaPlatform used by projects in VS Code extension. Maybe by changing the result of `JavaPlatformManager.getDefaultPlatform()`.

This is because the extension will/may require some relatively new JDK, but the projects may be developed for an older version.

I don't think anyone really tried that so far, though. (Jaroslav did some change in this direction for Maven long time ago, I think, but we need something more general.)

For source launcher, we could add a UI to select the JDK - it probably would not work in VS Code, but the same is true for JDK selection for other project types. If you want to look at that, you are of course welcome to. Or someone else might get to it, although I can't promise that, of course.

But I would prefer to avoid doing that here - this PR is complex enough on it own.

Thanks for the comment!

Reply:
> I think I (and possibly others) are considering separating the runtime JDK and the JavaPlatform used by projects in VS Code extension. Maybe by changing the result of JavaPlatformManager.getDefaultPlatform().

yes but since java files are not associated with a project, users can't even influence on which JDK they want to run the single/multi-file-java-program (SMFJP?). Which is independent of the default JDK vs nb runtime discussion. Controlling this over the default JDK is one way of doing this, but what if users don't want to run a file on the default JDK but on a specific JDK?

A JDK selector could be potentially added to the file properties, just like we have in maven/ant/gradle projects.
![SMFJP-jdk-selector](https://github.com/apache/netbeans/assets/114367/3a782f0a-9804-4cee-b494-d80a18a05bfe)